api/v1beta1/ovncontroller_types.go

```diff
@@ -119,14 +120,18 @@ func (instance OVNController) IsReady() bool {
 
 // OVSExternalIDs is a set of configuration options for OVS external-ids table
 type OVSExternalIDs struct {
-	SystemID  string `json:"system-id"`
-	OvnBridge string `json:"ovn-bridge"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="random"
+	SystemID  string `json:"system-id,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="br-int"
+	OvnBridge string `json:"ovn-bridge,omitempty"`
 
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default="geneve"
 	// +kubebuilder:validation:Enum={"geneve","vxlan"}
-	// OvnEncapType - geneve or vxlan
-	OvnEncapType string `json:"ovn-encap-type"`
+	OvnEncapType string `json:"ovn-encap-type,omitempty"`
```

Comment:
I would try to avoid omitempty if possible due to https://github.com/gibizer/operator-lint/tree/main/linters/crd/C003 

Unfortunately I only have a solution that needs some level of repetition of the default value:
https://github.com/gibizer/ovn-operator/commit/0c490b8305095ac7be97507249e2a0751c50ccb2

Interestingly it is enough to repeat one of the field defaults on the struct field level to get the rest automatically and it does not matter which one. (I still hate golang + kubebuilder json serialization logic)